tests/migrations/test_operations.py

```diff
@@ -4812,6 +4814,209 @@ def assertModelsAndTables(after_db):
+                        column_names=["author_id"],
+                    )
+                    yield
+                    actual_indexes = editor._constraint_names(
+                        Book,
+                        index=True,
+                        column_names=["author_id"],
+                    )
+                    self.assertNotEqual(set(actual_indexes), set(initial_indexes))
+                    # remove the author field as we should check creating missing
+                    # index each time
+                    recreate_author_field()
+                else:
+                    yield
+
+            def add_remove_unique_together(fields):
+                Book = project_state.apps.get_model(app_label, "Book")
+                original = Book._meta.unique_together
+                execute(
+                    migrations.AlterUniqueTogether("Book", list(original) + [fields]),
+                )
+                execute(migrations.AlterUniqueTogether("Book", original))
+
+            def add_remove_index_together(fields):
+                Book = project_state.apps.get_model(app_label, "Book")
+                original = Book._meta.index_together
+                execute(
+                    migrations.AlterIndexTogether("Book", list(original) + [fields]),
+                )
+                execute(migrations.AlterIndexTogether("Book", original))
+
+            def add_remove_constraint(constraint):
+                execute(migrations.AddConstraint("Book", constraint))
+                execute(migrations.RemoveConstraint("Book", constraint.name))
+
+            def add_remove_index(index):
+                execute(migrations.AddIndex("Book", index))
+                execute(migrations.RemoveIndex("Book", index.name))
+
+            unique_constraint = models.UniqueConstraint(
+                fields=["author", "title"],
+                name="author_title_uniq",
+            )
+            check_constraint = models.CheckConstraint(
+                check=models.Q(title__contains="a"),
+                name="title_check",
+            )
+
+            simple_index = models.Index(
+                fields=["author", "title"],
+                name="author_title_index",
+            )
+            f_index = models.Index(
+                models.F("author"),
+                models.F("title"),
+                name="author_title_index",
+            )
+            negative_index = models.Index(
+                fields=["-author", "title"],
+                name="author_title_index",
+            )
+            desc_index = models.Index(
+                models.F("author").desc(),
+                models.F("title"),
+                name="author_title_index",
+            )
+            lookup_index = models.Index(
+                models.F("title__lower"),
+                name="title_index",
+            )
+
+            actions_affect_indexes = [
+                lambda: add_remove_unique_together(["author", "title"]),
```

Comment:
This case works without the patch and it looks that it's already covered by `test_alter_unique_together()`. I'd remove it.